localstack/services/sqs/provider.py

```diff
@@ -795,6 +798,19 @@ def create_queue(
             # FIXME #5938: should raise `QueueNameExists` if queue exists with different attributes
             queue = backend.queues[queue_name]
             return CreateQueueResult(QueueUrl=queue.url(context))
+
+        # When you delete a queue, you must wait at least 60 seconds before creating a queue with the same name.
+        # see https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_DeleteQueue.html
+        if queue_name in backend.deleted and config.SQS_DELAY_RECENTLY_DELETED:
+            deleted = backend.deleted[queue_name]
+            if deleted > (time.time() - 60):
+                raise QueueDeletedRecently(
+                    "You must wait 60 seconds after deleting a queue before you can create another with the same name."
+                )
+            else:
+                del backend.deleted[queue_name]
```

Comment:
I am not sure about the lifecycle of the provider, but I think we are only deleting the entry in `self.deleted` if `config.SQS_DELAY_RECENTLY_DELETED` is `True`. But we are always adding the creation time when deleting in `delete_queue`. Not sure if important, but it could lead to memory leaks? We're using only the queue name as as key but the queue reference could be kept.

(Also, this is maybe my personal preference and I have no idea if it does change anything or if it's optimised internally afterwards, but I put the boolean check before the dictionary lookup to avoid it if the flag is not set. But I don't know if it's useful or not.)

Reply:
both great points :-)